packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.tsx

```diff
@@ -0,0 +1,160 @@
+    `,
+    buttonShow: css`
+      opacity: 1;
+      transition: opacity 0.1s ease;
+      z-index: 2;
+    `,
+    buttonHide: css`
+      opacity: 0;
+      transition: opacity 0.1s ease;
+      z-index: 0;
+    `,
+    confirmButtonContainer: css`
+      overflow: hidden;
+      position: absolute;
+      z-index: 1;
+    `,
+    confirmButton: css`
+      display: flex;
+      align-items: flex-start;
+    `,
+    confirmButtonShow: css`
+      opacity: 1;
+      transition: opacity 0.08s ease-out, transform 0.1s ease-out;
+      transform: translateX(0);
+    `,
+    confirmButtonHide: css`
+      opacity: 0;
+      transition: opacity 0.12s ease-in, transform 0.14s ease-in;
+      transform: translateX(100px);
+    `,
+  };
+});
+
+interface Props extends Themeable {
+  className?: string;
+  size?: ButtonSize;
+  confirmText?: string;
+  disabled?: boolean;
+  confirmVariant?: ButtonVariant;
+
+  onConfirm(): void;
+  onClick?(): void;
+  onCancel?(): void;
+}
+
+interface State {
+  showConfirm: boolean;
+}
+
+class UnThemedConfirmButton extends PureComponent<Props, State> {
+  static defaultProps: Partial<Props> = {
+    size: 'md',
+    confirmText: 'Save',
+    disabled: false,
+    confirmVariant: 'primary',
+  };
+
+  state: State = {
+    showConfirm: false,
+  };
+
+  onClickButton = (event: SyntheticEvent) => {
+    if (event) {
+      event.preventDefault();
+    }
+
+    this.setState({
+      showConfirm: true,
+    });
+
+    if (this.props.onClick) {
+      this.props.onClick();
+    }
+  };
+
+  onClickCancel = (event: SyntheticEvent) => {
+    if (event) {
+      event.preventDefault();
+    }
+    this.setState({
+      showConfirm: false,
+    });
+    if (this.props.onCancel) {
+      this.props.onCancel();
+    }
+  };
+
+  render() {
+    const {
+      theme,
+      className,
+      size,
+      disabled,
+      confirmText,
+      confirmVariant: confirmButtonVariant,
+      onConfirm,
+      children,
+    } = this.props;
+    const styles = getStyles(theme);
+    const buttonClass = cx(
+      className,
+      this.state.showConfirm ? styles.buttonHide : styles.buttonShow,
+      disabled && styles.buttonDisabled
+    );
+    const confirmButtonClass = cx(
+      styles.confirmButton,
+      this.state.showConfirm ? styles.confirmButtonShow : styles.confirmButtonHide
+    );
+    const onClick = disabled ? () => {} : this.onClickButton;
+
+    return (
+      <span className={styles.buttonContainer}>
+        {typeof children === 'string' ? (
+          <Forms.Button className={buttonClass} size={size} variant="link" onClick={onClick}>
+            {children}
+          </Forms.Button>
+        ) : (
+          <span className={buttonClass} onClick={onClick}>
+            {children}
+          </span>
+        )}
+        <span className={styles.confirmButtonContainer}>
+          <span className={confirmButtonClass}>
+            <Button size={size} variant="transparent" onClick={this.onClickCancel}>
```

Comment:
Why not using Forms.Button?

Reply:
`Forms.Button` has a blue border in a focused state, not sure it looks good in this context.

Reply:
That's by design/accessibility requirements. I think we should strive to use the new components if possible. Or maybe think about gentler way of indicating focus. @sarlinska @jessover9000 any thought on that?